homeassistant/util/__init__.py

```diff
@@ -261,6 +261,16 @@ def __init__(self, min_time, limit_no_throttle=None):
 
     def __call__(self, method):
         """Caller for the throttle."""
+        # Make sure we return a coroutine if the method is async.
```

Comment:
Pushed to the top so that throttling an async method with 2 throttle intervals will work.